README.md

```diff
@@ -90,6 +90,7 @@ Refer to this section for next steps when any indicator reports a `false`.
 |S0021|Determines if free memory is less than 10%.| Ensure your system hardware availablity matches the [recommended configuration](https://puppet.com/docs/pe/latest/hardware_requirements.html#hardware_requirements), note this assumes no third-party software using significant resources, adapt requirements accordingly for third-party requirements. | If you have issues with memory utilization in Puppet Enterprise that is not expected, open a Support ticket, referencing S0021 and provide the output of the [support script](https://puppet.com/docs/pe/latest/getting_support_for_pe.html#pe_support_script)
 | S0022        | Determines if there is a valid Puppet Enterprise license in place at `/etc/puppetlabs/license.key` on the primary server which is not expiring in the next 90 days.                | [Get help with Puppet Enterprise license issues](https://support.puppet.com/hc/en-us/articles/360017933313)                                                                                                                                                                                                                                                                                                                                                                              | Open a Support ticket referencing S0022 and provide the output of the following commands `ls -la /etc/puppetlabs/license.key` and `cat /etc/puppetlabs/license.key`. |
 | S0024        | Determines if there are files in the puppetdb discard directory newer than 1 week old                       |   Recent files indicate an issue that causes PuppetDB to reject incoming data. Invesitgate Puppetdb logs at the time the data was rejected to find a cause,                                                                                                                                                                                                                   | Open a Support ticket referencing S0024 and provide a copy of the PuppetDB log for the time in question, along with a sample of the most recent file in the following directory `/opt/puppetlabs/server/data/puppetdb/stockpile/discard/`
+| S0029        | Determines if number of concurrent connections to Postgresql DB is approaching 90% of the `max_connections` defined.  | Please look at `max_connections` at `/opt/puppetlabs/server/data/postgresql/11/data/postgresql.conf` and adjust it if needed. A restart is required to take effect.
```

Comment:
Actually, this needs to be changed as well.  This setting is managed by PE, so we can't just edit the file.  We also may not want to increase `max_connections`depending on why we're close to or exceeding the limit.  We would want to decrease the number of PDB read and write connections on compilers if a customer has a large number of them, for example.  I did a quick search for this in the KB and didn't see anything, so I created a ticket for it.

Reply:
Hi @m0dular, let me know how you want to update this.  As In Puppet the number of connections to Postgres is mostly determined by the sizes of the read and write pools in PDB.

puppet_enterprise::puppetdb::write_maximum_pool_size
puppet_enterprise::puppetdb::read_maximum_pool_size

Both of them are in turn determined by the **puppet_enterprise::puppetdb::command_processing_threads** 

**write_maximum_pool_size - "$command_processing_threads * 2"

read_maximum_pool_size - 
"puppet_enterprise::calculate_puppetdb_read_maximum_pool_size(\n                                                                  $command_processing_threads)"**

I tested with increase **max_connections** with restart of pe-postgresql.service, it went good, but should consider to increase **shared_buffers** if that is the case as each connection consumes RAM. 

Thanks.

Reply:
Sounds good.  I think for now until we finish writing the KB, we can just mention which settings to adjust.  Something like

> To increase the maximum number of connections in postgres, adjust `puppet_enterprise::profile::database::max_connections`.  Otherwise, to lower the number of connections created by PuppetDB, adjust x, y, and z.